src/grid/grid.component.ts

```diff
@@ -748,6 +751,10 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
 
     public unpinColumn(columnName: string) {
         const col = this.getColumnByName(columnName);
+
+        if (!col.pinned) {
+            return;
```

Comment:
It should return false when no unpinning is done. I noticed that at the end of the function there is no return true. That should also be added.